addons/test_mail/tests/test_mail_composer.py

```diff
@@ -1852,6 +1859,79 @@ def test_mail_composer_delete(self):
         self.assertEqual(len(self._new_msgs), 2, 'Should have created 1 mail.mail per record')
         self.assertFalse(self._new_msgs.exists(), 'Should have deleted mail.message records')
 
+    @users('employee')
+    @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_mail_composer_duplicates(self):
+        """ Ensures emails sent to the same recipient multiple times
+            are only sent when they are not duplicates
+        """
+        self.template.write({
+            'auto_delete': False,
+            'body_html': '<p>Common Body</p>',
+            'email_to': 'test@test.lan',
+            'subject': 'Common Subject',
+            'partner_to': '',
+        })
+
+        # Guarantee points of variation for test_report_3
+        self.test_records[0].write({'name': 'A', 'count': 1})
+        self.test_records[1].write({'name': 'B', 'count': 2})
+
+        attachment = self.env['ir.attachment'].create({
+            'datas': base64.b64encode(b'ExtraData'),
+            'mimetype': 'text/plain',
+            'name': 'SomeCommonAttachment.txt',
+            'res_model': False,
+            'res_id': False,
+        })
+
+        def _instanciate_composer():
+            composer_form = Form(self.env['mail.compose.message'].with_context(
+                self._get_web_context(self.test_records, add_web=True,
+                                      default_template_id=self.template.id)
+            ))
+            return composer_form.save()
+
+        base = {
+            'attachment_ids': [Command.clear()],
```

Comment:
Using 'Command' makes me very very sad.

Reply:
Why must you insist on having the wrong opinion :sob:

Reply:
Note that normally you should follow the file's global styling, which is old school 2014 openerp code... so I am not totally wrong.